File: brainbox/find_words_containing_character.py
def findWordsContaining(words, x):
    result = []
    for i, w in enumerate(words):
        if x in w:
            result.append(i)
    return result


# ---- Manual Tests ----
print(findWordsContaining(["leet","code"], "e"))          # [0, 1]
print(findWordsContaining(["abc","bcd","aaaa","cbc"], "a"))  # [0, 2]
print(findWordsContaining(["abc","bcd","aaaa","cbc"], "z"))  # []
print(findWordsContaining([], "a"))                        # []
print(findWordsContaining(["x"], "x"))                     # [0]
print(findWordsContaining(["hello","world"], "l"))         # [0, 1]
print(findWordsContaining(["aaa","bbb","ccc"], "b"))       # [1]

File: brainbox/find_words_containing_character.js
// -------------------------
// Solution
// -------------------------
function findWordsContaining(words, x) {
  const result = [];
  for (let i = 0; i < words.length; i++) {
    if (words[i].includes(x)) {
      result.push(i);
    }
  }
  return result;
}

// -------------------------
// Tests
// -------------------------
const tests = [
  { words: ["leet", "code"], x: "e", expected: [0, 1] },
  { words: ["abc", "bcd", "aaaa", "cbc"], x: "a", expected: [0, 2] },
  { words: ["hello", "world"], x: "z", expected: [] },
  { words: ["x", "xx", "abc"], x: "x", expected: [0, 1] },
  { words: ["apple", "banana", "kiwi"], x: "i", expected: [2] },
];

tests.forEach(({ words, x, expected }) => {
  const result = findWordsContaining(words, x);
  console.log(
    `words=${JSON.stringify(words)}, x="${x}" → ${JSON.stringify(result)} (expected ${JSON.stringify(expected)})`
  );
});
